src/handlers/miner_hbbft_handler.erl

```diff
@@ -262,6 +260,39 @@ handle_message(<<BinMsgIn/binary>>, Index, #state{hbbft=HBBFT}=S0) ->
         {ok, {signature, _, Addr, Sig}} ->
             handle_sigs({received, [{Addr, Sig}]}, S0);
 
+        {ok, {proposed_skip, ProposedRound, IndexRound}} when ProposedRound =/= CurRound ->
+            lager:info("proposed skip to round ~p", [ProposedRound]),
+            case process_skips(ProposedRound, IndexRound, S0#state.f, Index, Skips) of
+                {skip, Skips1} ->
+                    lager:info("skipping"),
+                    %% ask for some time
+                    miner:reset_late_block_timer(),
+                    SkipGossip = {multicast, t2b({proposed_skip, ProposedRound, CurRound})},
+                    %% skip but don't discard rounds until we get a clean round
+                    {HBBFT1, ToSend1} =
+                        case hbbft:next_round(HBBFT, ProposedRound, []) of
+                            {H1, ok} ->
+                                {H1, []};
+                            {H1, {send, NextMsgs}} ->
+                                {H1, NextMsgs}
+                        end,
+                    {HBBFT2, ToSend2} =
+                        case hbbft:start_on_demand(HBBFT1) of
+                            {H2, already_started} ->
+                                {H2, []};
+                            {H2, {send, Msgs}} ->
+                                {H2, Msgs}
+                        end,
+                    ToSend = fixup_msgs(ToSend1 ++ ToSend2),
+                    {(state_reset(HBBFT2, S0))#state{skip_votes = Skips1},
```

Comment:
Why do we retain the skip votes here?

Reply:
we retain them until we complete a block, in case we fail and new members join and we converge on a different round.